src/storage/src/region.rs

```diff
@@ -107,13 +108,14 @@ impl<S: LogStore> Region for RegionImpl<S> {
 ///
 /// Contains all necessary storage related components needed by the region, such as logstore,
 /// manifest, memtable builder.
-pub struct StoreConfig<S> {
+pub struct StoreConfig<S: LogStore> {
```

Comment:
Could we remove this trait bound on the config struct? So we don't always need a LogStore to construct the StoreConfig.

Reply:
This trait bound is introduced by `CompactionRequestImpl`, `CompactionRequestImpl` has a field `Wal<S>` where `S: LogStore`.